lib/identity_cache/cache_key_generation.rb

```diff
@@ -9,14 +9,16 @@ def self.schema_to_string(columns)
 
     def self.denormalized_schema_hash(klass)
       schema_string = schema_to_string(klass.columns)
-      klass.send(:all_cached_associations).sort.each do |name, options|
+      klass.send(:all_cached_associations).sort.each do |name, association|
         klass.send(:check_association_scope, name)
-        ParentModelExpiration.check_association(options) if options[:embed]
-        case options[:embed]
-        when true
-          schema_string << ",#{name}:(#{denormalized_schema_hash(options[:association_reflection].klass)})"
-        when :ids
+        ParentModelExpiration.check_association(association) if association.embedded?
+        case association
+        when Cached::Recursive::Association
+          schema_string << ",#{name}:(#{denormalized_schema_hash(association.reflection.klass)})"
+        when Cached::Reference::HasMany
           schema_string << ",#{name}:ids"
+        when Cached::Reference::HasOne
+          schema_string << ",#{name}:id"
```

Comment:
There's no has_one :id case here originally, so I'll fix that in another PR before merging this.